qiskit_experiments/library/hamiltonian/heat_zx.py

```diff
@@ -0,0 +1,220 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+HEAT experiments for ZX Hamiltonian.
+"""
+
+from typing import Tuple, Optional
+
+import numpy as np
+from qiskit.circuit import QuantumCircuit
+from qiskit.providers import Backend
+
+from .heat_base import BatchHeatHelper, HeatElement
+from .heat_analysis import HeatAnalysis
+
+
+class ZXHeat(BatchHeatHelper):
+    """HEAT experiment for the ZX-type entangler.
+
+    # section: overview
+        This experiment is designed to amplify the error contained in the
+        ZX-type generator, a typical Hamiltonian implemented
+        by a cross resonance drive, which is a foundation of the CNOT gate.
+
+        The echo circuit refocuses ZX rotation to identity (II) then applies
+        a pi-pulse along the interrogated error axis. X error and Y error are
+        amplified outward the X-Y plane to draw a ping-pong pattern with
+        state flip by the pi-pulse echo, while Z error is amplified outward
+        the X-Z plane in the same manner.
+        Measurement is projected onto Y-axis in this setup.
+        Because the echoed axis are anti-commute with other Pauli terms,
+        errors in other axes are cancelled out to reduce rotation in the interrogated axis.
+        This enables to selectively amplify the Hamiltonian dynamics in the specific axis.
+        Note that we have always nonzero X rotation imparted by the significant ZX term,
+        the error along Y and Z axis are skewed by the nonzero commutator term.
+        This yields slight mismatch in the estimated coefficients with the generator Hamiltonian,
+        however this matters less when the expected magnitude of the error is small.
+        On the other hand, the error in the X axis is straightforward
+        because this is commute with the ZX term of the generator.
+
+        .. parsed-literal::
+                             (xN)
+                             ░ ┌───────┐                            ░
+            q_0: ────────────░─┤0      ├────────────────────────────░─────────
+                 ┌─────────┐ ░ │  heat │┌────────────────┐┌───────┐ ░ ┌───┐┌─┐
+            q_1: ┤ Rα(π/2) ├─░─┤1      ├┤ Rx(-1.0*angle) ├┤ Rβ(π) ├─░─┤ γ ├┤M├
+                 └─────────┘ ░ └───────┘└────────────────┘└───────┘ ░ └───┘└╥┘
+            c: 1/═══════════════════════════════════════════════════════════╩═
+                                                                            0
```

Comment:
I wonder if we can make the prep, meas etc easier to see, e.g.:
```suggestion
        .. parsed-literal::
             prep                heat            echo                 meas
                             (xN)
                             ░ ┌───────┐                            ░
            q_0: ────────────░─┤0      ├────────────────────────────░─────────
                 ┌─────────┐ ░ │  heat │┌────────────────┐┌───────┐ ░ ┌───┐┌─┐
            q_1: ┤ Rα(π/2) ├─░─┤1      ├┤ Rx(-1.0*angle) ├┤ Rβ(π) ├─░─┤ γ ├┤M├
                 └─────────┘ ░ └───────┘└────────────────┘└───────┘ ░ └───┘└╥┘
            c: 1/═══════════════════════════════════════════════════════════╩═
                                                                            0
```

Reply:
this is excellent idea!